libflux/src/flux/semantic/bootstrap.rs

```diff
@@ -248,7 +344,7 @@ mod tests {
     use crate::semantic::parser::parse;
 
     #[test]
-    fn infer_program() {
+    fn infer_program() -> Result<(), Error> {
```

Comment:
Yes `main` functions as well as test functions can return results. And yes it will fail if it does not return `Ok(())`. Note the error type must implement `Debug` as that is what will be used to display output in the case of test failure. This allows one to use the `?` operator in `main` and test functions.